tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/StaticFiles/DirectoryBrowserExtensionsUseDirectoryBrowserIntegration.cs

```diff
@@ -0,0 +1,57 @@
+// <copyright file="DirectoryBrowserExtensionsUseDirectoryBrowserIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Iast;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore.StaticFiles;
+
+/// <summary>
+/// Microsoft.AspNetCore.Builder.IApplicationBuilder Microsoft.AspNetCore.Builder.DirectoryBrowserExtensions::UseDirectoryBrowser(Microsoft.AspNetCore.Builder.IApplicationBuilder) calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "Microsoft.AspNetCore.StaticFiles",
+    TypeName = "Microsoft.AspNetCore.Builder.DirectoryBrowserExtensions",
+    MethodName = MethodName,
+    ReturnTypeName = "Microsoft.AspNetCore.Builder.IApplicationBuilder",
+    ParameterTypeNames = ["Microsoft.AspNetCore.Builder.IApplicationBuilder"],
+    MinimumVersion = "2",
+    MaximumVersion = "8",
+    IntegrationName = nameof(IntegrationId.AspNetCore))]
+[InstrumentMethod(
+    AssemblyName = "Microsoft.AspNetCore.StaticFiles",
+    TypeName = "Microsoft.AspNetCore.Builder.DirectoryBrowserExtensions",
+    MethodName = MethodName,
+    ReturnTypeName = "Microsoft.AspNetCore.Builder.IApplicationBuilder",
+    ParameterTypeNames = ["Microsoft.AspNetCore.Builder.IApplicationBuilder", "Microsoft.AspNetCore.Builder.DirectoryBrowserOptions"],
+    MinimumVersion = "2",
+    MaximumVersion = "8",
+    IntegrationName = nameof(IntegrationId.AspNetCore))]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public class DirectoryBrowserExtensionsUseDirectoryBrowserIntegration
+{
+    private const string MethodName = "UseDirectoryBrowser";
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<DirectoryBrowserExtensionsUseDirectoryBrowserIntegration>();
+
+    internal static CallTargetReturn<TReturn?> OnMethodEnd<TTarget, TReturn>(TReturn? returnValue, Exception? exception, in CallTargetState state)
```

Comment:
Just wondering why you do this `OnMethodEnd` instead of `OnMethodStart`? Is it a sneaky way to handle multiple arguments? 😅 Does it matter that you don't record it until _after_ it's already leaked?

Reply:
Yes and no.
Yes, it's less code to handle that on `OnMethodEnd` to skip the arguments.
No, it doesn't matter to record the vulnerability after, because that's a call that set up the directory browser (that is done at the start of the app, so no connection has been made yet).